### ISSUE TITLE
Type checking ignores script-setup blocks

### ISSUE THREAD

**Issue by bodograumann** [vuejs/vue-cli#6716]
### Version
5.0.0-beta.4

### Reproduction link
[github.com](https://github.com/bodograumann/vue-cli-script-setup-lang-ts-typechecks)

### Environment info
```
  System:
    OS: Linux 5.14 Arch Linux
    CPU: (4) x64 Intel(R) Core(TM) i5-7200U CPU @ 2.50GHz
  Binaries:
    Node: 16.10.0 - /usr/bin/node
    Yarn: 1.22.4 - ~/.local/bin/yarn
    npm: 7.24.1 - /usr/bin/npm
  Browsers:
    Chrome: Not Found
    Firefox: 92.0.1
  npmGlobalPackages:
    @vue/cli: 5.0.0-beta.4
```

### Steps to reproduce
```
vue create vue-cli-script-setup-lang-ts-typechecks -bi '{
      "useConfigFiles": false,
      "plugins": {
        "@vue/cli-plugin-typescript": {
          "classComponent": false
        }
      },
      "vueVersion": "3"
    }'
cd vue-cli-script-setup-lang-ts-typechecks
```

Add some wrongly typed code. E.g.:
```
echo '<script setup lang="ts">const x: string = 42;</script>' >>> src/App.vue
```

Build
```
npm run build
```

### What is expected?
The typescript error is detected and the build fails.

### What is actually happening?
The typing is simply stripped and no error can be seen.

**Comment by curly210102** [comment]
@sodatea  This issue and #6736 are related to `fork-ts-checker-webpack-plugin`, ts-checker doesn't support `<script setup>` yet, I've submitted an issue and 

But think a little further, is it better to create Vue3 project with [vue-cli-plugin-vite](https://github.com/IndexXuan/vue-cli-plugin-vite)? Vite can provide long-term support for Vue3.

**Comment by batuotasis** [comment]
any progress on this?

**Comment by bodograumann** [comment]
Please update to https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/releases/tag/v6.4.1 where the issue is fixed.

**Comment by screetBloom** [comment]
ok, i will try it

**Comment by screetBloom** [comment]
Ok, then this issue can be closed

**Comment by bodograumann** [comment]
No. While it worked for me, others are still reporting uncovered edge cases.
I think we should do the update in vue-cli, but this issue needs to stay open until everything is covered.

**Comment by screetBloom** [comment]
it worked for me too, then it can be closed first, for unknown case maybe  can add another issue 😄

**Comment by victorgarciaesgi** [comment]
Hi! Sorry to reopen, I have this problem with Vue 3 and script setup not showing typescript errors on console. I tried forcing fork-ts-checker version with pnpm but no changes. Did you manage to resolve it?

**Comment by bodograumann** [comment]
The first try to solve this in fork-ts-checker-webpack-plugin was reverted and no further attempt has been made afaict.
I have switched to vite since and am using vue-tsc there anway. I think vue-tsc also works in vue-cli context as a workaround. Just has to be triggered manually. @victorgarciaesgi

**Comment by victorgarciaesgi** [comment]
Thanks @bodograumann ! Decided to use `vue-tsc --watch` for dev mode too it works great!
